### ISSUE TITLE
influxdb 2.6.1 - v1 API - unable to filter tags with WHERE after restart

### ISSUE THREAD

**Issue by mateuszdrab** [influxdata/influxdb#24148]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Deployed a brand new instance of influxdb from bitnami charts
2. Created my data bucket
3. Created  `v1`  read and write accounts with respective read and write permission to the bucket
4. Fed my data (about 600 measurements) to influxdb using a python script (using the write account)
5. Queried the data using Grafana (using the read account)
6. At this point all was working fine, I considered the project complete
7. A new helm chart version was installed resulting in the pod restarting, after the restart, I was not able to query as below

__Expected behavior:__
The query:
`SELECT mean("consumption") FROM "electricity" WHERE ("account_name" = 'Account') AND $timeFilter GROUP BY time(30m) fill(linear)`

__Actual behavior:__
Before restart, this works fine - I get results.
After restart, getting no results.
The data can be browsed through the web UI before and after restart.
Removing the `account_name` condition and adding GROUP BY account_name indeed shows me the results along with the respective account_name which I can no longer filter via the WHERE clause.

I can get it to work again by wiping the bucket contents with `influx delete --bucket octograph --start 2023-01-01T00:00:00Z --stop 2024-01-01T00:00:00Z` and re-feeding the data.

This leads me to believe some kind of cache or index is lost.

__Environment info:__

* System info: Run `uname -srm` and copy the output here
* InfluxDB version: Run `influxd version` and copy the output here
* Other relevant environment details: Container runtime, disk info, etc
Data path is `/bitnami/influxdb` in the bitnami build.
Ls of the directory shows:
`configs  data  influxd.bolt  influxd.sqlite  lost+found  replicationq  wal`

__Config:__
Default values, I made no significant customizations so the configuration is whatever the chart creates with the below values:
```
  values:
    influxdb:
      service:
        type: LoadBalancer
      updateStrategy:
        type: Recreate
    metrics:
      enabled: true
      serviceMonitor:
        enabled: true
    auth:
      existingSecret: influxdb
```
` /opt/bitnami/influxdb/etc/influxdb.conf` is empty

__Logs:__
No errors in logs, query just returns no results.

**Comment by jeffreyssmith2nd** [comment]
We don't maintain the bitnami charts, so I don't have much familiarity with them. A couple of questions:

1. Is this easily reproducible?
2. If so, can you reproduce it outside of K8s using one of our supported releases?
^ Should help determine if it is an issue with how the bitnami charts are built.

**Comment by mateuszdrab** [comment]
> We don't maintain the bitnami charts, so I don't have much familiarity with them. A couple of questions:
> 
> 1. Is this easily reproducible?
All I have to do to reproduce it is to restart the influxd process in the container
> 2. If so, can you reproduce it outside of K8s using one of our supported releases?
>    ^ Should help determine if it is an issue with how the bitnami charts are built.
My plan is to try that next/or an older version of influxdb

**Comment by jeffreyssmith2nd** [comment]
One other thing would be to see if an equivalent Flux query works:

```
SELECT mean("consumption") FROM "electricity" WHERE ("account_name" = 'Account') AND $timeFilter GROUP BY time(30m) fill(linear)
```

should turn into something along the lines of:

```
from(bucket: "electricity")
  |> range(start: <timeFilter.start>, stop: <timeFilter.stop>)
  |> filter(fn: (r) => r["account_name"] == "Account")
  |> aggregateWindow(every: 30m, fn: mean, createEmpty: false)
  |> yield(name: "consumption")
```

**Comment by mateuszdrab** [comment]
@jeffreyssmith2nd 

Thanks, I had to adjust the query a bit - however, it does yield results.
The only thing is those results are further 'grouped by' additional tags that my influxql query would ignore

![image](https://user-images.githubusercontent.com/26556541/226392172-3d68f44e-49cc-4526-80b3-8fb8668e2fc7.png)

I guess I have two paths: switch to Flux or dig deeper to figure out what's happening with InfluxQL queries.

**Comment by jeffreyssmith2nd** [comment]
I think your InfluxQL query is fine, my Flux was just meant to be a loose guideline based on what I could tell of your schema.

So the flux query worked across reboots as well?

**Comment by mateuszdrab** [comment]
Interestingly, my script injected another day of consumption data into the bucket and I can query (with InfluxQL)  the new data (past day), but not the old data that was already there.

There must be some index or something not being loaded properly when influxdb is restarted.

**Comment by jeffreyssmith2nd** [comment]
A quick scan doesn't show anything that immediately jumps out to me, if I have some spare cycles I can take a deeper look.

You may also want to try running just their image itself. That would help isolate the chart vs the image. I would also consider standing up a second instance of the chart if possible to test to see if it is actually reproducible or if there is something odd about the data in the existing cluster.

If you find that it is not reproducible anywhere else, the `influxd inspect` commands (especially `dump-tsi`, `dump-tsm`, `report-tsi` and `report-tsm`) may point out if there's a data issue. You can also use `build-tsi` to rebuild the indexes, but I would only do that if you have solid backups in place first.

From the bitnami README on how to run their image standalone.
```
docker run \
    --volume /path/to/influxdb-persistence:/bitnami/influxdb \
    --env INFLUXDB_HTTP_AUTH_ENABLED=false \
    bitnami/influxdb:latest
```

**Comment by mateuszdrab** [comment]
I did not manage to repro it in the Bitnami container so, I stopped the cluster deployment and removed the volume and recreated it, created the bucket, v1 user again and fed the data, queried - it worked, removed the pod and waited for the new one to start, queried, it worked again... 

This is a bit bizarre, I guess I should have wiped it earlier and started again

The only thing that wonders me is that I renamed the bucket a few times - but overall it was a brand new instance I created a few days ago. I don't get how this issue didn't get carried over when I copied the database from the pod.

Thanks for your time though, if I reproduce it again I will re-open the issue.

**Comment by mateuszdrab** [comment]
I did not manage to repro it in the Bitnami container so, I stopped the cluster deployment and removed the volume and recreated it, created the bucket, v1 user again and fed the data, queried - it worked, removed the pod and waited for the new one to start, queried, it worked again...

This is a bit bizarre, I guess I should have wiped it earlier and started again

The only thing that wonders me is that I renamed the bucket a few times - but overall it was a brand new instance I created a few days ago. I don't get how this issue didn't get carried over when I copied the database from the pod.

Thanks for your time though, if I reproduce it again I will re-open the issue.
